### ISSUE TITLE
Pluggable allocator weighting policies + drawdown-aware gross exposure

### ISSUE THREAD

**Issue by jugegp16** [jugegp16/midas#32]
## Background

#29 replaced the per-asset sigmoid+normalize allocator with a softmax construct-to-budget design. Targets now sum to investable by construction and every trade attributes to a real conviction strategy. The current softmax treats all tickers as risk-equivalent: a Momentum score of +0.5 produces the same target weight whether the ticker is VOO (~14% annualized vol) or IBIT (~70% annualized vol). For a single-asset-class momentum book this is fine. For Midas's actual portfolio (`portfolio.yaml` — 21 tickers spanning broad-market ETFs, sector ETFs, mega-cap tech, speculative growth, and crypto, with a ~10x vol spread across the universe) this means equal-conviction tilts produce wildly unequal risk contributions.

This issue covers two independent improvements that fit together as a single PR:

1. **Pluggable weighting policies** — make the per-ticker score modifier inside the softmax a configurable strategy, with `equal` (current behavior) as the default and `inverse_vol` as the second option. This is the architectural slot that any future weighting scheme (target-vol, ERC, min-variance, Black-Litterman) plugs into.
2. **Drawdown-aware gross exposure scaling** — independent knob that scales `investable` based on realized portfolio drawdown. CPPI-style risk control. Universal, applies regardless of weighting policy.

## Why pluggable, not hard-coded inverse-vol

Different portfolio shapes want different weighting:

- A 3-ticker single-sector momentum book (like `sample-portfolio.yaml`) probably wants `equal` because momentum's edge is largest on volatile names — inverse-vol would systematically dampen the highest-conviction signals.
- A diversified multi-asset portfolio (like `portfolio.yaml`) probably wants `inverse_vol` because vol spread is large and real diversification exists, so risk-balanced exposure matters more than per-asset conviction tilt.
- A future user with bonds + commodities + FX added would want `inverse_vol` or even full ERC.

Making it configurable is how every mature portfolio construction library works: LEAN's `PortfolioConstructionModel` is a polymorphism point, PyPortfolioOpt's `EfficientFrontier` / `HRPOpt` / `CLA` / `BlackLittermanOpt` are interchangeable, cvxportfolio's `SinglePeriodOpt` takes a pluggable objective. Hard-coding any specific weighting forces philosophy on the user.

### Part 1: Pluggable weighting policies

Add a `weighting` field to the allocator config:

```yaml
allocator:
  weighting: equal              # default — current softmax behavior
  # or
  weighting: inverse_vol
  vol_lookback_days: 20         # only used when weighting != equal
  vol_floor: 0.005              # clamp to avoid degenerate low-vol blowups
```

Implementation slots into the existing softmax. Each policy contributes a per-ticker score offset that gets added inside the exponent:

```
target_i = budget * exp(k * blended_i + offset_i) / sum_j(exp(k * blended_j + offset_j))
```

- `equal`: `offset_i = 0` for all tickers (reduces to current behavior exactly).
- `inverse_vol`: `offset_i = -log(vol_i)` (mathematically identical to dividing the exponent by vol).

The construct-to-budget invariant survives (`sum(targets) == budget` exactly) for any choice of offsets. Cap-with-redistribution (Phase 4) re-runs softmax over survivors with the same offsets — composes naturally. Option A (neutral=hold) is unchanged: held tickers are excluded from the softmax entirely and their weight comes from `current_weights`.

Vol is precomputed once in `Allocator.precompute_signals` alongside the signal cache, indexed by `len(prices) - 1` for cheap lookup during `allocate()`. Realized close-to-close log-return vol over `vol_lookback_days`, annualized by `sqrt(252)`. EWMA / GARCH variants can come later as additional policies.

**Warmup handling**: tickers with fewer than `vol_lookback_days` of data are treated as Option A holds (no vol estimate, hold current weight). Avoids the "5 days of flat history → vol≈0 → softmax pushes ticker to ~budget" failure mode.

### Part 2: Drawdown-aware gross exposure

Independent knob, applies regardless of weighting policy:

```yaml
allocator:
  drawdown_scaling:
    enabled: true
    penalty: 1.5     # exposure_scale = max(1 - penalty * current_dd, floor)
    floor: 0.5       # never scale gross below 50% of investable
```

Mechanic: track running peak portfolio value, compute current drawdown, scale `investable` by `max(1 - penalty * current_dd, floor)` before passing into the softmax. The freed budget becomes deliberate cash reserve. Recovers automatically as portfolio heals.

This is standard CPPI / Kelly-fraction risk control. Bridgewater, Man AHL, Winton, and most institutional trend followers do some form of this. It's universal — works with any weighting policy, any strategy mix, any portfolio shape — and it bounds tail risk on portfolios where vol spread makes drawdowns brutal (the sample backtest in #29 had a 48.55% max drawdown, which is on the wrong side of acceptable for most users).

The current portfolio value comes from the same place the rebalancer computes it for `current_weights`. Peak tracking lives in the backtest/live engine state, passed into `allocate()` as a single float (current_drawdown).

## Scope

- `src/midas/allocator.py`: WeightingPolicy enum/class, `_compute_offsets`, vol cache in `precompute_signals`, drawdown scaling in `allocate()`. ~150 lines.
- `src/midas/models.py`: new fields on `AllocationConstraints` (or a nested `WeightingConfig`). ~10 lines.
- `src/midas/backtest.py`: track peak value, compute current_drawdown, pass to allocator. ~15 lines.
- `src/midas/live.py`: same as backtest. ~10 lines.
- `src/midas/optimizer.py`: expose `vol_lookback_days`, `drawdown_penalty`, `drawdown_floor` as searchable hyperparameters. ~20 lines.
- Tests: weighting policy reduces to softmax for equal, concentrates on low-vol for inverse_vol, vol floor protects degenerate input, vol respects warmup, drawdown scaling reduces investable on losses and recovers, sum-to-budget invariant survives across all policies. ~150 lines.
- `docs/architecture.md`: update Phase 2 + Phase 4 sections, add weighting-policy subsection. ~30 lines.

Estimated 350-400 lines total.

## Success criteria

- [ ] `weighting: equal` reproduces current behavior bit-for-bit (regression test)
- [ ] `weighting: inverse_vol` on `portfolio.yaml` produces dramatically different allocations than `equal` (concentrate on low-vol tickers when conviction is similar)
- [ ] Drawdown scaling reduces gross exposure during simulated drawdown periods and recovers
- [ ] All existing tests pass without modification
- [ ] Walk-forward optimizer can search the new hyperparameters
- [ ] Backwards-compatible: existing YAMLs without the new fields behave exactly as today

## Out of scope

- Hierarchical allocation tree (separate issue)
- Correlation-aware weighting / ERC / full Markowitz (later, after pluggable policy infrastructure exists)
- Vol forecasting beyond simple realized close-to-close (EWMA, GARCH, Parkinson — later)
- Per-ticker target-vol normalization (different objective; could become a separate weighting policy)

## Related

- Builds on #29 (softmax allocator)
- Will be referenced by the hierarchical-tree issue (the pluggable policy slot is what each tree node will use)

**Comment by jugegp16** [comment]
Folded into two issues: pluggable weighting policies (Part 1) → #53, drawdown-aware gross exposure scaling (Part 2) → #52. Both target issues updated with the merged content. Closing as duplicate.
